chore(atomic): migrate atomic-commerce-breadbox

alexprudhomme opened the pull request:
Have another PR. for the functional component noise https://github.com/coveo/ui-kit/pull/5474

# atomic-commerce-breadbox Component Specification
📝 Requirements
The atomic-commerce-breadbox component is designed to display a summary of currently active facet selections in the form of breadcrumbs. It allows users to understand which filters are applied and to interactively remove individual or all filters.

Display active filters: Each facet selection (breadcrumb) is shown with a label and value(s), formatted based on the facet type (e.g., regular, numeric, date, hierarchical).

Clear filters: Users can remove individual filters or clear all filters at once.

Show more / show less: When breadcrumbs overflow the visible area, users can expand or collapse the list.

Truncate long hierarchical paths: Paths can be truncated to a user-defined pathLimit, replacing the middle segments with an ellipsis.

Responsive layout: Breadcrumb visibility adapts to available container width using ResizeObserver.

♿ Accessibility
Uses the FocusTargetController to manage focus after filter removal or breadcrumb expansion/collapse, ensuring good keyboard navigation and screen reader support.

Buttons (clear, show more/less, breadcrumb removal) include aria-labels for screen readers.

The breadcrumb list is rendered semantically as an unordered list , and each breadcrumb is a list item, enhancing navigability.

Show more/less buttons are conditionally shown/hidden based on content overflow and provide accessible descriptions.

⚡ Performance & Security
Performance:

Uses ResizeObserver to adapt the breadcrumb display based on available width.

Collapsed breadcrumb logic hides elements with display: none to prevent unnecessary rendering without removing them from the DOM.

Lazy rendering behavior avoids rendering when no breadcrumbs exist.


⚠️ Risks and Challenges
Dynamic layout complexity: Managing breadcrumb overflow with custom logic and DOM measurement (e.g., scrollWidth > clientWidth) can be brittle across browsers or container styles.

Focus management: Ensuring proper focus targets and keyboard interactions across different breadcrumb states (collapsed, expanded, cleared) requires careful coordination.

Facet type variations: Supporting various facet types (e.g., numeric, date, hierarchical) adds complexity, especially in value formatting and truncation logic.


---

## ✅ Checklist

- [x] 🧪 The component is [unit tested](https://docs.google.com/document/d/1lrsHAUROGMkRbsGlor02eYKF2mfrfBa0SOESxHFU3pI/edit?tab=t.0#heading=h.l9yzzrwf1i4k)
- [x] 🧪 The component includes [E2E tests](https://docs.google.com/document/d/1lrsHAUROGMkRbsGlor02eYKF2mfrfBa0SOESxHFU3pI/edit?tab=t.0#heading=h.z9tw07qe1lai)
- [x] 🗑️ Old Cypress tests exclusive to the component are removed
- [x] 📖 The component is documented in storybook with an `.mdx` file
- [x] ♿ The component complies with the [Web Content Accessibility Guidelines](https://www.w3.org/TR/WCAG21/).
- [x] 🌐 All strings intended for humans or assistive technology must be localized with i18n.
- [x] 📦 The Lit component is exported in the appropriate `index.ts` and `lazy-index.ts` files.
- [x] 🎨 CSS parts are documented still accessible.
- [x] 🦥 Slotted Content, public methods and properties are documented
- [x] 🔄 The component outputs the same Angular output as before with Stencil
- [x] 🏷️ The component declares the component type in the HTMLElementTagNameMap

https://coveord.atlassian.net/browse/KIT-4251

alexprudhomme commented on packages/atomic/src/utils/date-utils.ts:
unused param
https://github.com/coveo/ui-kit/pull/5478

lvu285 commented:
Few comments I have

#1. Breadbox doesn’t collapse when it’s too long. You will need to refresh the page in order to make the breadbox collapsed
![image](https://github.com/user-attachments/assets/07d8bbfe-43a9-4310-93e6-ab0303aa675c)
 

#2. pathLimit looks like this parameter doesn’t work, for example I set


#3. Uncheck value from Breadbox list doesn’t work
![image](https://github.com/user-attachments/assets/832a8648-ff51-410d-aba7-7ca6655270cf)


#4. I’m not sure how CSS-path breadcrumb-list-container & breadcrumb-list works, but when I set like below, it doesn’t change anything
``` atomic-commerce-breadbox::part(breadcrumb-list-container) {
        color: red;
      }```
 

#5. question mark is it expected that the Css-path for container and label are the same? as I changed the color, only the Filter-label changed in both case


#6.  I’m not sure if it’s expected or not, but on the Main atomic-search with Stencil, the Category facet with many children, we display like this _ with ... at the end
![image](https://github.com/user-attachments/assets/9239d31f-2cbb-41b6-9e62-e68552b27743)

But with commerce facet we have ... in the middle. 
![image](https://github.com/user-attachments/assets/ef9be97e-21bc-4cc9-b6e8-6e291aeb8991)
 

#7. Not sure if it’s related or not, or maybe related to some works for Facets that somehow happens on your branch…

Issue: if I select around 9-10 facets value (for example under Brand), when selecting it works fine, results display ok. But then if I refresh the page, I will get into Something wrong with console log

```BAD_REQUEST - The Feign client's server 'http://SearchApi' returned status code '400' with server header 'null' and error message : '{
  "statusCode" : 400,
  "message" : "Number of non-idle values of the facet 'ec_brand' must be less than or equal to 'numberOfValues'",
  "type" : "NonIdleValueCountGreaterThanNumberOfValuesException",
  "executionReport" : [ {
    "children" : [ {
      "name" : "RequestId",
      "description" : "The id correlating logs for this request",
      "duration" : 0,
      "X-Request-ID" : "2df0b29a-08dc-4772-8b6f-c646cce19358"```

alexprudhomme commented:
> Few comments I have
> 
> #1. Breadbox doesn’t collapse when it’s too long. You will need to refresh the page in order to make the breadbox collapsed ![image](https://private-user-images.githubusercontent.com/21175826/456649278-07d8bbfe-43a9-4310-93e6-ab0303aa675c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAyNzM1MTEsIm5iZiI6MTc1MDI3MzIxMSwicGF0aCI6Ii8yMTE3NTgyNi80NTY2NDkyNzgtMDdkOGJiZmUtNDNhOS00MzEwLTkzZTYtYWIwMzAzYWE2NzVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjE4VDE5MDAxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEwYzNlMzU4MWM4OWQzMDQ5YTUxMzJhNDMwMTFiNWJiYmM3M2MwNmQxYzhiYTMyMGZlNGQzN2ZlNzg2MTljNTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Tb912fslkeqUOC1bWHOxh79z4t4Aj6JFZz9dAjQnYdY)
> 
> ....

1. Fixed and added a test
2. we talked about pathLimit
3. Fixed and added a test
4 & 5. These are the three parts and what they represent. Seems fine to me !
<img width="657" alt="image" src="https://github.com/user-attachments/assets/00770444-df13-4f3a-ba4f-30f597c345fb" />

<img width="743" alt="image" src="https://github.com/user-attachments/assets/928f3d4e-7e6b-493f-b56e-3abd9ab52d23" />
6. Normal behavior of pathLimit
7. this is a headles bug. I am looking into it

fpbrault reviewed (approved):
LGTM!

lvu285 reviewed (approved):
Looks good on my side!
